docs/transforms.md

```diff
@@ -0,0 +1,116 @@
+
+## Adding and configuring a transform
+
+Since a transform is a type of processor, all transforms are created through the `processors` endpoint. 
+Here's an example:
+
+```json lines
+POST /v1/processors
+{
+    // name of the transform
+    "name": "extractfieldpayload",
+    // type of processor: TYPE_TRANSFORM or TYPE_FILTER 
+    "type": "TYPE_TRANSFORM",
+    "parent": 
+    {
+        // type of parent: TYPE_CONNECTOR or TYPE_PIPELINE
+        "type": "TYPE_CONNECTOR",
+        // parent ID (connector ID in this case)
+        "id": "aed07589-44d8-4c68-968c-1f6c5197f13b" 
+    },
+    "config":
+    {
+        "settings":
+        {
+            // configuration map for this processor
+            "field": "name" 
+        }
+    }
+}
+```
+The request to create a processor is described in [api.swagger.json](/pkg/web/openapi/swagger-ui/api/v1/api.swagger.json).
+
+## Supported transforms
+
+Conduit provides a number of built-in transforms, such as filtering fields, replacing them, posting payloads to HTTP endpoints etc.
+Conduit also provides the ability to write custom transforms in JavaScript.
+
+### Built-in transforms
+
+An up-to-date list of all built-in transforms and detailed descriptions can be found [here](https://pkg.go.dev/github.com/conduitio/conduit/pkg/processor/transform/txfbuiltin).
+An example is available in [extract-field-transform.sh](/examples/transforms/extract-field-transform.sh). The script will
+set up a pipeline with the built-in extract-field transform.
+
+### JavaScript transforms
+
+JavaScript transforms make it possible to write custom transforms. The API name for JavaScript transforms (used in the
+request to create a transform) is `js`. There's only one configuration parameter, `script`, which is the script itself.
+
+The script needs to implement a function called `transform`, which accepts an `sdk.Record`, and returns an `sdk.Record`:
+```javascript
+function transform(record) {
+    // do something with the record
+    return record
+}
+```
+
+The script, of course, is not constrained to having only this function, i.e. you can have something like this:
+```javascript
+function doSomething(record) {
+    // do something with the record
+    return record
+}
+
+function transform(record) {
+    doSomething(record)
+    return record
+}
+```
+
+Conduit also provides a number of helper objects and methods which can be used in the JS code. Those are, currently:
+1. `logger` - a `zerolog.Logger` which writes to the Conduit server logs
+2. `Record()` - constructs a `record.Record`.
+3. `RawData()` - constructs `record.RawData`.
+
+All the helpers are defined in [helpers.go](/pkg/processor/transform/txfjs/helpers.go). Following is an example of a JavaScript
+transform:
+```javascript
+// Parses the record payload as JSON
+function parseAsJSON(record) {
+    return JSON.parse(String.fromCharCode.apply(String, record.Payload.Bytes()))
+}
+
+function transform(record) {
```

Comment:
@hariso Do I understand correctly that `transform` is the entry point for javascript functions? 

@lovromazgon have we thought about parity between our `turbine` frameworks and the primitives used there vs what we've chosen to do here? I'm thinking about `process` in turbine.

Reply:
You are right, `transform` is the entrypoint right now, this was designed before Turbine even existed. That said, I agree with you, we should probably revisit this design. AFAIK @hariso is working on a refactoring that will simplify processors and remove the notion of transforms/filters - maybe as part of that we can rename `tranform` to `process`, that way we move to a more intuitive name given that we will only have "processors" and we achieve parity with Turbine.

Reply:
@neovintage That's correct, `transform` is the entry point for JS functions. The name can be changed at zero cost, so if we come up with a different name, it's doable. 

I can chime in a little bit about the primitives. We considered the following options (together with @jmar910 to get some insight into Turbine JS too):
1. Processor with type TRANSFORM, and a `transform` JS function **and** a processor with type FILTER.
The processor type feels a redundant piece of information, so we were looking for ways to drop it.
2. Processor (no type defined) with a single JS function, `transform` (or whatever we call it). The actual action performed by Conduit (transform or filter/drop record) is decided based on the output: if the function returns `null`, the record is dropped.
3.  Processor (no type defined), with two functions: `transform` and `filter`. This would make it analogous with `map` and `filter` used on JS arrays (and Turbine does work on arrays of records). However, James' opinion was that it would actually make it more confusing.

Reply:
My vote goes to option 2, sounds simple and straightforward.

Reply:
FTR, we discussed the 3rd option too in details in a call, and went for option 2 for the reasons Lovro mentioned. I'll explain the details in docs.